### PR TITLE
update touchin generamba template

### PR DESCRIPTION
**Pull request by kopytovs**

### CODE REVIEW

File: touchin/Code/ViewModel/view_model.swift.liquid [hunk 1]
```diff
@@ -1,8 +1,34 @@
 import LeadKit
 import RxSwift
 
-final class {{ module_info.name }}ViewModel: BaseViewModel {
+// MARK: - Constants
 
-    private let disposeBag = DisposeBag()
+private enum Constants {
```
Comment (petropavel13) [inline touchin/Code/ViewModel/view_model.swift.liquid]: remove

File: touchin/Code/View/view_controller.swift.liquid [hunk 2]
```diff
@@ -3,49 +3,77 @@ import LeadKit
 import RxSwift
 import RxCocoa
 
-final class {{ module_info.name }}ViewController: UIViewController {
+// MARK: - Constants
 
-    let viewModel: {{ module_info.name }}ViewModel
+private enum Constants {
```
Comment (petropavel13) [inline touchin/Code/View/view_controller.swift.liquid]: remove

File: touchin/Code/View/view_controller.swift.liquid [hunk 3]
```diff
@@ -3,49 +3,77 @@ import LeadKit
...
+}
 
-    init(viewModel: {{ module_info.name }}ViewModel) {
-        self.viewModel = viewModel
-        super.init(nibName: nil, bundle: nil)
-    }
+final class {{ module_info.name }}ViewController: BaseViewController<{{ module_info.name }}ViewModel> {
```
Comment (petropavel13) [inline touchin/Code/View/view_controller.swift.liquid]: BaseConfigurableController

File: touchin/Code/View/view_controller.swift.liquid [hunk 4]
```diff
@@ -3,49 +3,77 @@ import LeadKit
...
+    }
+
+}
+
+// MARK: - Action
+
+private extension {{ module_info.name }}ViewController {
+}
+
+// MARK: - Constraints
```
Comment (petropavel13) [inline touchin/Code/View/view_controller.swift.liquid]: did you mean public?

File: touchin/Code/View/view_controller.swift.liquid [hunk 5]
```diff
@@ -3,49 +3,77 @@ import LeadKit
...
 
+    override func configureBarButtons() {
+        super.configureBarButtons()
+
+        <#code#>
+    }
+
+}
+
+// MARK: - Action
```
Comment (petropavel13) [inline touchin/Code/View/view_controller.swift.liquid]: what for?
objc actions will be inside class

File: touchin/Code/ViewModel/view_model.swift.liquid [hunk 6]
```diff
@@ -1,8 +1,34 @@
...
+    // MARK: - Life Cycle
+
+}
+
+//MARK: - Actions
 
+extension {{ module_info.name }}ViewModel {
 }
+
+//MARK: - Public
```
Comment (petropavel13) [inline touchin/Code/ViewModel/view_model.swift.liquid]: add space

File: touchin/Code/ViewModel/view_model.swift.liquid [hunk 7]
```diff
@@ -1,8 +1,34 @@
...
+
+    // MARK: - Computed Properties
+
+    // MARK: - Services
+
+    // MARK: - Life Cycle
+
+}
+
+//MARK: - Actions
```
Comment (petropavel13) [inline touchin/Code/ViewModel/view_model.swift.liquid]: smae

File: touchin/Code/ViewModel/view_model.swift.liquid [hunk 8]
```diff
@@ -1,8 +1,34 @@
...
 
+extension {{ module_info.name }}ViewModel {
 }
+
+//MARK: - Public
+
+extension {{ module_info.name }}ViewModel {
+}
+
+//MARK: - Private
```
Comment (petropavel13) [inline touchin/Code/ViewModel/view_model.swift.liquid]: same

File: touchin/Code/ViewModel/view_model.swift.liquid [hunk 9]
```diff
@@ -1,8 +1,34 @@
...
+
+final class {{ module_info.name }}ViewModel: BaseControllerViewModel {
+
+    // MARK: - Properties
+
+    // MARK: - Computed Properties
+
+    // MARK: - Services
+
+    // MARK: - Life Cycle
```
Comment (petropavel13) [inline touchin/Code/ViewModel/view_model.swift.liquid]: life cycle for view model?

File: touchin/Code/ViewModel/view_model.swift.liquid [hunk 10]
```diff
@@ -1,8 +1,34 @@
...
 import RxSwift
 
-final class {{ module_info.name }}ViewModel: BaseViewModel {
+// MARK: - Constants
 
-    private let disposeBag = DisposeBag()
+private enum Constants {
+}
+
+final class {{ module_info.name }}ViewModel: BaseControllerViewModel {
```
Comment (petropavel13) [inline touchin/Code/ViewModel/view_model.swift.liquid]: BaseViewModel

**Review by iON1k — APPROVED** [review]

**Review by petropavel13 — CHANGES_REQUESTED** [review]

File: touchin/Code/ViewModel/view_model.swift.liquid (hunk 9 above)
Comment (kopytovs) [reply touchin/Code/ViewModel/view_model.swift.liquid]: meant `init`

File: touchin/Code/View/view_controller.swift.liquid [hunk 11]
```diff
@@ -3,49 +3,69 @@ import LeadKit
...
+    // MARK: - Views
 
-    required init?(coder aDecoder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
-    }
+    // MARK: - Properties
 
     // MARK: - View Life Cycle
 
     override func viewDidLoad() {
```
Comment (petropavel13) [inline touchin/Code/View/view_controller.swift.liquid]: remove

File: touchin/Code/View/view_controller.swift.liquid (hunk 4 above)
Comment (petropavel13) [reply touchin/Code/View/view_controller.swift.liquid]: rename to public

File: touchin/Code/View/view_controller.swift.liquid [hunk 12]
```diff
@@ -3,49 +3,69 @@ import LeadKit
 import RxSwift
 import RxCocoa
 
-final class {{ module_info.name }}ViewController: UIViewController {
+final class {{ module_info.name }}ViewController: BaseConfigurableController<{{ module_info.name }}ViewModel>, DisposeBagHolder {
```
Comment (petropavel13) [inline touchin/Code/View/view_controller.swift.liquid:6]: DisposeBagHolder is not implemented

**Review by petropavel13 — CHANGES_REQUESTED** [review]

File: touchin/Code/View/view_controller.swift.liquid [hunk 13]
```diff
@@ -3,49 +3,66 @@ import LeadKit
...
-        super.init(nibName: nil, bundle: nil)
-    }
+    // MARK: - Views
 
-    required init?(coder aDecoder: NSCoder) {
-        fatalError("init(coder:) has not been implemented")
-    }
+    // MARK: - Properties
+
+    var disposeBag = DisposeBag()
```
Comment (petropavel13) [inline touchin/Code/View/view_controller.swift.liquid]: private let

**Review by petropavel13 — CHANGES_REQUESTED** [review]

**Review by petropavel13 — APPROVED** [review]

**Pull request merged**